accessibility/common-remediations.md

```diff
@@ -0,0 +1,200 @@
+
+Where a built-in interactive element is available, use it. 
+
+We do this:
+```html
+<button>Toggle UI</button>
+<a href=”/”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<div onclick=”doFunc();”>Toggle UI</div>
+<span>Visit this location</a>
+```
+
+
+#### Links with no href
+
+All anchor links (the `a` element) require an `href` attribute in order to be functional as links. Omitting the `href` attribute prevents keyboard users and voice input users from being able to use the link at all. 
+
+We do this:
+```html
+<a href=”#location”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<a onclick=”myFunc();”>Visit this location</a>
+```
+
+
+
+####  Inappropriate negative tabindex
+
+Interactive elements with a negative tabindex attribute (`tabindex=-1`) are not usable by keyboard users. 
+
+
+
+### Visual order is not followed with source code
+Just try to follow the visual order of elements while coding in HTML, don’t adjust elements to take positions with CSS, then this is confusing to keyboard users who expect to tab forwards and backwards in a linear sequence.
+
+As a rule, try tabbing through your pages every so often just to make sure you haven't accidentally messed up the tab order. 
+
+We do this:
+```html
+<button> I Should <button />
+<button> Be focused<button />
+<button> Last <button />
+```
+
+We don't do this:
+```html
+<button style=”float: right”> I Should <button />
+<button> Be focused<button />
+<button> Last <button />
+```
+
+
+
+### No advance warning when opening new window 
+
+Opening new windows automatically when a link is activated can be disorienting for people who have difficulty perceiving visual content, and for some people with cognitive disabilities, if they are not warned in advance.
+
+We do this:
+```html
+<a href="https://mysite.com" target="_blank" rel="noopener">
+ My site
+ <span class="visuallyhidden">, opens in a new window</span>
+</a>
+```
+
+We don’t do this:
+```html
+<a href="knitting.html" target="_blank">All about Knitting</a>
+```
+
+
+
+## Images
+
+### Images with no alt attribute
+
+Alt attributes are required on all `img` tags. Your HTML is invalid without them. 
+
+We do this:
+```html
+<img src="/images/NYCpark.png" alt="Aerial view of Central Park in New York" />
+```
+
+We don't do this:
+```html
+<img src="/images/NYCpark.png" />
+```
+
+
+### Meaningful images with meaningless alt attributes
+
+Alt text should be used to convey the intent of an image. Avoid using generic strings like “photo”, “image”, or “icon” as alt values, as they don’t communicate valuable content to the user. Be as descriptive as possible.
+
+We do this:
+```html
+<img src="/images/kinghenry.jpg" alt="King Henry VIII of England" />
```

Comment:
```suggestion
<img src="/images/kinghenry.jpg" alt="King Henry VIII of England">
```
We implicitly close void elements - https://github.com/springernature/frontend-playbook/blob/master/markup/house-style.md#implicitly-close-void-elements

(I did all of these code examples and I put the closing `/` in there in all of them, it's all my fault 🤦‍♀ )